Marlin/Regulator.cpp

```diff
@@ -33,4 +69,98 @@ void setOutputPressure(float desired_pressure) {
     DAC_write(MCP4725_I2C_ADDRESS, digital_val);
 }
 
+
+/**
+ * Updates the protection function, called regularly in the main loop.
+ */
+void Regulator__Update() {
+  pressure_protection(current_target_pressure, pressureRegulator());
+}
+
+//===========================================================================
+//============================ Private Functions ============================
+//===========================================================================
+
+/** 
+ * Error handler when marlin detects a missing pressure regulator
+ */
+static void _regulator_sled_removed_error() {
+  if (sled_present == true) {
```

Comment:
It seems to me that the checking of `sled_present` should be done outside of the error function. If this function is ever called, it should execute the body.